lib/edda-core/src/api_types/rebuild_request.rs

```diff
@@ -0,0 +1,213 @@
+mod v1;
+
+use std::{
+    fmt,
+    ops::{Deref, DerefMut},
+};
+
+use naxum_api_types::{ApiVersionsWrapper, ApiWrapper, RequestId, UpgradeError};
+use serde::{Deserialize, Serialize};
+use strum::{AsRefStr, EnumDiscriminants, EnumIs, EnumString, VariantNames};
+
+pub use self::v1::RebuildRequestV1;
+
+pub type RebuildRequestVCurrent = RebuildRequestV1;
+
+#[derive(Clone, Eq, Serialize, PartialEq, VariantNames)]
+#[serde(rename_all = "camelCase")]
+#[strum(serialize_all = "camelCase")]
+pub enum RebuildRequest {
+    V1(RebuildRequestV1),
+}
+
+impl ApiWrapper for RebuildRequest {
+    type VersionsTarget = RebuildRequestVersions;
+    type Current = RebuildRequestVCurrent;
+
+    const MESSAGE_TYPE: &'static str = "RebuildRequest";
+
+    fn id(&self) -> RequestId {
+        match self {
+            Self::V1(RebuildRequestVCurrent { id, .. }) => *id,
+        }
+    }
+
+    fn new_current(current: Self::Current) -> Self {
+        Self::V1(current)
+    }
+}
+
+impl fmt::Debug for RebuildRequest {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::V1(inner) => inner.fmt(f),
+        }
+    }
+}
+
+impl Deref for RebuildRequest {
+    type Target = RebuildRequestVCurrent;
+
+    fn deref(&self) -> &Self::Target {
+        match self {
+            Self::V1(inner) => inner,
+        }
+    }
+}
+
+impl DerefMut for RebuildRequest {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        match self {
+            Self::V1(inner) => inner,
+        }
+    }
+}
+
+#[remain::sorted]
+#[derive(Clone, Debug, Deserialize, EnumDiscriminants, EnumIs, Eq, PartialEq, VariantNames)]
+#[serde(rename_all = "camelCase")]
+#[strum(serialize_all = "camelCase")]
+#[strum_discriminants(strum(serialize_all = "camelCase"), derive(AsRefStr, EnumString))]
+pub enum RebuildRequestVersions {
+    V1(RebuildRequestV1),
+}
+
+impl ApiVersionsWrapper for RebuildRequestVersions {
+    type Target = RebuildRequest;
+
+    fn id(&self) -> RequestId {
+        match self {
+            Self::V1(RebuildRequestV1 { id, .. }) => *id,
+        }
+    }
+
+    fn into_current_version(self) -> Result<Self::Target, UpgradeError> {
+        match self {
+            Self::V1(inner) => Ok(Self::Target::V1(inner)),
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use std::{
+        error::Error,
+        fmt,
+        fs::File,
+        io::{self, BufRead as _, BufReader, Read as _},
+        path::Path,
+    };
+
+    use serde::{de::DeserializeOwned, Serialize};
+
+    use super::RebuildRequestVersionsDiscriminants;
+
+    /// Tests that a versioned object will always serialize to the same representation, no matter
+    /// what future versions or changes to the object.
+    ///
+    /// NOTE: It is imperative that incrementatl refactorings do not lead to a version-incompatible
```

Comment:
while we're talking comment typos: incrementatl, the next outkast single

Reply:
Fletcher! 🇨🇦